### PR TITLE
Add android docker and docker compose files.

### PR DESCRIPTION
**Pull request by jrbenny35**
https://mozilla-hub.atlassian.net/browse/ECTEEN-77

This adds an docker file for running the android tests. 

This is an example command:
```docker compose -f docker-compose-android.yml run -it -e EXPERIMENT_SLUG="android-add-ons-extension-process" -e PYTEST_ARGS="-m smoke_test" klaatu-android```

### CODE REVIEW

File: Dockerfile.android
Comment (pdehaan) [inline Dockerfile.android:1]: Does this file need a copyright header at all?

File: Dockerfile.android [hunk 1]
```diff
@@ -0,0 +1,38 @@
+
+FROM python:3.11.3-alpine
```
Comment (pdehaan) [inline Dockerfile.android:2]: I'm not a big Python person, but should this be 3.11.x or 3.12?
Per python.org:
> Latest: [Python 3.12.1](https://www.python.org/downloads/release/python-3121/)

File: Dockerfile.desktop [hunk 2]
```diff
@@ -0,0 +1,97 @@
+FROM ubuntu:lunar
+
+USER root
+
+# Set env
+ENV DEBIAN_FRONTEND=noninteractive \
+    MOZ_HEADLESS=1 \
+    LC_ALL=C.UTF-8 \
+    LANG=C.UTF-8 \
+    FIREFOX_VERSION=112.0 \
```
Comment (pdehaan) [inline Dockerfile.desktop:10]: Why so old? Per https://whattrainisitnow.com/,

VERSION | CHANNEL
----:|----
115| [ESR](https://whattrainisitnow.com/release/?version=esr)
120 | [Release](https://whattrainisitnow.com/release/?version=release)
121 | [Beta](https://whattrainisitnow.com/release/?version=beta)
122 | [Nightly](https://whattrainisitnow.com/release/?version=nightly)

And it also looks like tomorrow is release day, so numbers will bump again:
> _"Firefox 121 ships on December 19 (in 1 day)"_

File: Dockerfile.desktop [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    GECKODRIVER_VERSION=0.33.0
```
Comment (pdehaan) [inline Dockerfile.desktop:11]: 👍  https://github.com/mozilla/geckodriver/releases

File: Dockerfile.desktop [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    && apt-get -y purge firefox \
+    && rm -rf /var/lib/apt/lists/* /var/cache/apt/*
+
+ENV VIRTUAL_ENV=/opt/venv
+RUN python3 -m venv $VIRTUAL_ENV
+ENV PATH="$VIRTUAL_ENV/bin:$PATH"
+
+# Install Firefox and Geckodriver
+RUN INSTALLER_DOWNLOAD_URL=https://raw.githubusercontent.com/hackebrot/install-firefox/master/install-firefox.sh \
+    && curl $INSTALLER_DOWNLOAD_URL -sSf | sh -s -- --firefox $FIREFOX_VERSION --geckodriver $GECKODRIVER_VERSION
```
Comment (pdehaan) [inline Dockerfile.desktop:45]: So the `FIREFOX_VERSION` and `GECKODRIVER_VERSION` from this file, override the contents of the GitHub repo above on L44?
> ```ini
> FIREFOX_VERSION="59.0"
> GECKODRIVER_VERSION="0.20.0"
> ```

File: Dockerfile.android
Comment (jrbenny35) [reply Dockerfile.android:1]: It probably should

File: Dockerfile.android (hunk 1 above)
Comment (jrbenny35) [reply Dockerfile.android:2]: I'll have to do this later as there are some changes needed upstream in the android repo

**Review by pdehaan — APPROVED** [review]
R+, but looks like all the GitHub checks are failing. The one I randomly spot checked was saying:
> git@github.com: Permission denied (publickey).
> fatal: Could not read from remote repository.

**Comment by jrbenny35** [comment]
I think I need to configure circleci again and update the checks

**Pull request merged**